backend-api/tests/api-tests/clientCredentialsGrantFlow.test.ts

```diff
@@ -42,14 +58,110 @@ describe("POST /token", () => {
       expect(response.status).toBe(400);
     });
   });
+
+  describe("Given the grant_type value is invalid", () => {
+    it("Returns a 400 Bad Request response", async () => {
+      const response = await axiosInstance.post(
+        `${apiBaseUrl}/async/token`,
+        "grant_type=invalid_grant_type",
+        {
+          headers: {
+            "x-custom-auth": "Basic " + toBase64(clientIdAndSecret),
+          },
+        },
+      );
+
+      expect(response.data).toStrictEqual({
+        error: "invalid_grant",
+        error_description: "Invalid grant type or grant type not specified",
+      });
+      expect(response.status).toBe(400);
+    });
+  });
+
+  describe("Given there is no Authorization header in the request", () => {
+    it("Returns a 400 Bad Request response", async () => {
+      const response = await axiosInstance.post(
+        `${apiBaseUrl}/async/token`,
+        "grant_type=client_credentials",
+      );
+
+      expect(response.data).toStrictEqual({
+        error: "invalid_authorization_header",
+        error_description: "Invalid authorization header",
+      });
+      expect(response.status).toBe(400);
+    });
+  });
+
+  describe("Given the client in the Authorization header is not a registered client", () => {
+    it("Returns a 400 Bad Request response", async () => {
+      const invalidClientIdAndSecret = "invalidClient:invalidClientSecret";
+      const response = await axiosInstance.post(
+        `${apiBaseUrl}/async/token`,
+        "grant_type=client_credentials",
+        {
+          headers: {
+            "x-custom-auth": "Basic " + toBase64(invalidClientIdAndSecret),
+          },
+        },
+      );
+
+      expect(response.data).toStrictEqual({
+        error: "invalid_client",
+        error_description: "Supplied client credentials not recognised",
```

Comment:
API spec says this `error_description` should be `Invalid client` - how do we want to proceed?